include/board.h

```diff
@@ -0,0 +1,58 @@
+#include <stddef.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+// setup
+void init_board(void);
+
+// clock
+uint32_t millis();
+uint64_t micros();
+
+// serial
+void serial_init(uint32_t baud_rate);
+void serial_write(uint8_t byte);
+uint16_t serial_bytes_available(void);
+uint8_t serial_read(void);
+
+// sensors
+void sensors_init(int board_revision);
+
+void imu_register_callback(void (*callback)(void));
+void imu_read_accel(float accel[3]);
```

Comment:
Yeah I wasn't sure about that... It might eliminate a couple of unnecessary copies. Maybe I'll reevaluate once I get the build stuff figured out